### DISCUSSION TITLE
Naming scheme peculiarities

### DISCUSSION THREAD

**Discussion by rrrnld** [jimsalterjrs/sanoid#823] [General]
I have zfs snapshots on multiple datasets that I want sanoid to consider as its own. My understanding of https://github.com/jimsalterjrs/sanoid/discussions/800#discussioncomment-4910366 is that it should be enough to rename those to match sanoid's scheme. So I rename snapshots with the following naming scheme:

```
data1/sftp/staging@post-backup--2023-05-15T00:00:12Z
```

to sanoid's scheme like so:

```
data1/sftp/staging@autosnap_2023-05-15_00:00:12_daily
```

My sanoid config looks like this:

```
[template_staging]
        frequently = 0
        hourly = 0
        daily = 30
        monthly = 12
        yearly = 5
        autosnap = yes
        autoprune = yes

[data1]
	recursive = zfs
	use_template = staging
```

I rename the snapshot first, then run sanoid. It still creates a new daily snapshot. This is what it looks like for the last two days:

```
data1/sftp/staging@autosnap_2023-05-13_00:00:13_daily
data1/sftp/staging@autosnap_2023-05-13_21:15:10_daily
data1/sftp/staging@autosnap_2023-05-14_00:00:11_daily
data1/sftp/staging@autosnap_2023-05-14_21:15:23_daily
```

Is there a misunderstanding on my side? I'd like to create monthly and yearly snapshots if necessary, but consider the daily snapshots that were taken before and prune them according to the rules given in the config file. Is this possible? Thanks!

**Comment by ovizii** [comment]
Did you ever figure this out? oh, and btw. how did you rename the snapshots? Is there a way to mass-rename somehow?

**Reply by rrrnld** [reply]
I wrote a custom script that lists snapshots via `zfs list -r -t snapshot $DATASET | grep pattern` and renames them one by one

**Comment by phreaker0** [comment]
@heyarne the problem is that you used "recursive = zfs" instead of "recursive = yes".

With "recursive = zfs" zfs handle the recursion (which is atomic), so sanoid will only look for snaphosts on the root dataset of the recursion (which is data1 in your case) for deciding if a snapshot needs to be taken.

**Reply by ovizii** [reply]
Is that documented somewhere? I can't seem to find any info about this fact.

**Reply by phreaker0** [reply]
only in the configuration file https://github.com/jimsalterjrs/sanoid/blob/a7e6c2db680a76784448ba1f810d7df9a50c3ddc/sanoid.conf#L31

**Reply by rrrnld** [reply]
Am I at risk of losing snapshots if I change the setting to `recursive = yes`?

**Reply by phreaker0** [reply]
No, pruning is done the same way with recursive = zfs and recursrive = yes, according to your configured prune policy.
